src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSignature.DSA.cs

```diff
@@ -145,7 +145,7 @@ protected override bool VerifyKeyType(AsymmetricAlgorithm key)
                     {
                         var signature = new ReadOnlySpan<byte>(rented, 0, bytesWritten);
 
-                        if (key != null && !certificate.GetDSAPublicKey().VerifySignature(dataHash, signature))
+                        if (key != null && !certificate.GetDSAPublicKey()!.VerifySignature(dataHash, signature))
```

Comment:
What guarantees GetDSAPublicKey() here won't return null?

Reply:
Something way closer to caller code only got here because everything said it was DSA.